distro/otel.go

```diff
@@ -69,10 +73,16 @@ func Run(opts ...Option) (SDK, error) {
 		// TODO: configure batching policy with configured values.
 		trace.WithSpanProcessor(trace.NewBatchSpanProcessor(exp)),
 	)
-	global.SetTracerProvider(traceProvider)
+	otel.SetTracerProvider(traceProvider)
 
-	// TODO: add and honor option to set additional propagators.
-	global.SetTextMapPropagator(b3.New())
+	if c.Propagator != nil {
+		if c.Propagator == nonePropagator {
+			// Set to an empty propagator if none was specified
+			otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator())
+		} else {
```

Comment:
how about implementing `nonePropagatorType` as a NullObject (noop implementation) so that this `if` statement would not be required?

Reply:
By making `nonePropagatorType` a no-op implementation and then setting the global propagator to that value it would contradict the behavior of what setting `OTEL_PROPAGATORS` to `none` is intended to accomplish (i.e. no automatically configured propagator).

Reply:
Would it change anything for the user? What is the difference between an empty composite propagator or a composite propagator with a noop propagator.

Reply:
Moreover take notice that even in https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#enum-value the `none` is described as "a null object (empty, no-op) value"

Reply:
> Moreover take notice that even in https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#enum-value the `none` is described as "a null object (empty, no-op) value"

Sure, but please take note of [the specific specification for `OTEL_PROPAGATOR=none`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#general-sdk-configuration):

> `none`: No automatically configured propagator.

Hence the omission of automatically configuring a propagator.

Reply:
I see. Feel free to resolove is setting  OTEL_PROPAGATORS=tracecontext,none **does not** result in a crash at runtime